services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/service/impl/DpmServiceImpl.java

```diff
@@ -373,7 +383,15 @@ private void patchGoalstateForNeighbor(UnicastGoalState unicastGoalState) throws
                 //unicastGoalState.setHostIp(neighborInfo.getHostIp());
                 unicastGoalState.setHostIp(hostIp);
                 unicastGoalState.getGoalStateBuilder().addNeighborStates(neighborState);
-                patchGoalstateForNeighbor(unicastGoalState);
+                UnicastGoalState unicastGoalStateTemp = new UnicastGoalState();
+                unicastGoalStateTemp.getGoalStateBuilder().addNeighborStates(neighborState);
+                patchGoalstateForNeighbor(networkConfig, unicastGoalStateTemp);
```

Comment:
This block of code appears to be very customized. Was a UnicasetGoalStateTemp 100% necessary? If it would, let us put comment to explain the context.

Reply:
I have put comments in the code.
`unicastGaolStateTemp` is to get patchGoalStates (subnet_states and a consolidated router_state) based on the current neighborState.

Reply:
Thanks James. Is UnicastGoalStateTemp reusable? If it could be created once and reused by multiple entry in the for loop, it could save us something.